### PR TITLE
Addendum to #672

### PR DESCRIPTION
**Pull request by cevich**
Fixes several test failures on the latest/greatest RHEL w/ docker-1.12.5.

The 'display failing output check in build test' commit depends on #672 being merged first.

### CODE REVIEW

File: config_defaults/subtests/docker_cli/negativeusage.ini [hunk 1]
```diff
@@ -66,8 +66,8 @@ stderr = Invalid hostPort
 
 [docker_cli/negativeusage/iv4]
 subcmd = run
-subarg = -e,PATH=/tmp,%%(FQIN)s,true
-stderr = exec\: \"true\"\: executable file not found in \$PATH
+subarg = -e,PATH=/tmp,%%(FQIN)s,tRuE
+stderr = .*tRuE.*not found
```
Comment (edsantiago) [inline config_defaults/subtests/docker_cli/negativeusage.ini]: This masks bz1393572: we will never know if/when it is fixed, nor if it breaks again. Is this your intention?

File: subtests/docker_cli/build/build.py [hunk 2]
```diff
@@ -208,22 +209,26 @@ def parse_postprocess_commands(self, build_def):
...
-            # Verify zero exit status + healthy output
-            if OutputGood(build_def.dockercmd.cmdresult,
-                          ignore_error=True):
+            # Verify zero exit status and healthy output
+            opg = OutputGood(build_def.dockercmd.cmdresult,
+                             ignore_error=True)
+            if opg:
                 return build_def.dockercmd.cmdresult.exit_status == 0
             else:
+                self.loginfo(str(opg))
```
Comment (edsantiago) [inline subtests/docker_cli/build/build.py]: Should this be `warn` (vs info)?

Also, given that there's an identical log below, would you consider adding a differentiator to each one? E.g.

    self.logwarning("OutputGood failed: %s" % opg)
....
    self.logwarning("OutputNotBad failed: %s" % opb)

That makes it easier for a human to determine which of those two logs triggered, and continue debugging from there.

File: subtests/docker_cli/build/build.py [hunk 3]
```diff
@@ -208,22 +209,26 @@ def parse_postprocess_commands(self, build_def):
...
                 return False
         elif command == 'negative':
-            # Verify non-zero exit status + healthy output
-            if OutputGood(build_def.dockercmd.cmdresult,
-                          ignore_error=True,
-                          skip=['error_check']):
+            # Verify non-zero exit status and no panics
+            opb = OutputNotBad(build_def.dockercmd.cmdresult,
```
Comment (edsantiago) [inline subtests/docker_cli/build/build.py]: I'm really uncomfortable with the name `opb`, especially when contrasted to `opg`. I read opb as 'output bad', not 'output NOT bad'. A casual reader may make the same assumption. How about just renaming both opg and opb to `ok`?

File: config_defaults/subtests/docker_cli/negativeusage.ini (hunk 1 above)
Comment (cevich) [inline config_defaults/subtests/docker_cli/negativeusage.ini]: No it wasn't.  I had a feeling there was something more to the breakage here.  Thanks for the ref. BZ.  I think a better "fix" would be for me to just stick that BZ in a comment and drop all the other changes.

File: subtests/docker_cli/build/build.py (hunk 2 above)
Comment (cevich) [inline subtests/docker_cli/build/build.py]: good idea.

File: subtests/docker_cli/build/build.py (hunk 3 above)
Comment (cevich) [inline subtests/docker_cli/build/build.py]: %s/opb/opng/g

**Comment by cevich** [comment]
@edsantiago Thanks.  All addressed and force-pushed.

File: subtests/docker_cli/build/build.py [hunk 4]
```diff
@@ -208,22 +209,28 @@ def parse_postprocess_commands(self, build_def):
...
+                self.logwarn('Positive output expected but check failed: %s',
+                             str(opg))
                 return False
         elif command == 'negative':
-            # Verify non-zero exit status + healthy output
-            if OutputGood(build_def.dockercmd.cmdresult,
-                          ignore_error=True,
-                          skip=['error_check']):
+            # Verify non-zero exit status and no panics
+            opng = OutputNotBad(build_def.dockercmd.cmdresult,
```
Comment (edsantiago) [inline subtests/docker_cli/build/build.py]: That's still confusing! If the function is *NotBad*, I find a variable named *ng* misleading. Did you mean to make it *nb* ?

Comment (cevich) [inline subtests/docker_cli/build/build.py]: > Did you mean to make it nb ?

Uggg, yes...The number of letters has no impact on my ability to spell correctly.  Thankyou edautospellchecker 😀 

(really, I did mean ``opnb`` but my head is elsewhere)

Comment (cevich) [inline subtests/docker_cli/build/build.py]: oops, there's a bug in the printing part too (wrong variable)

**Pull request merged**